models/articles.model.js

```diff
@@ -36,4 +37,24 @@ exports.fetchAllArticles = () => {
     return db.query(query).then(({rows}) => {
         return rows;
     });
+}
+
+exports.updateVotesByArticleId = (id, update) => {
+    
+    const votesQuery = (id, update) => {
+        return db.query(`
+            UPDATE articles
+            SET votes = votes + $2
+            WHERE article_id = $1
+            RETURNING *;
+            `, [id, update.inc_votes])
+        .then(({rows}) => rows[0])
+    }
+
+
+    return Promise.all([votesQuery(id, update), this.fetchArticleById(id)])
```

Comment:
Why have you used the `this` keyword here?

Reply:
We don't need a Promise.all here. We don't need to check the article exists because there is less ambiguity here. If we get an empty array, it means the article does not exist

Reply:
got rid of this